### PR TITLE
Redo PR5

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Part of #1

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
I am not sure why the test cases are failing. Though passing in my local. Moreover,

File: extractor/src/test/java/com/example/extractor/MainTest.java [hunk 1]
```diff
@@ -32,8 +32,11 @@ public void testGeneratedConfigMatchesExpectedExample1() throws Exception {
         String exampleFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example1.java";
         String templateFilePath = "src/main/resources/config-template-treewalker.xml";
         String expectedFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Configs/config-example1-expected.xml";
+        String outputFilePath = "src/test/resources/generated-config-example1.xml";
```
[the remark this answers is not in the document]
Comment (relentless-pursuit) [reply extractor/src/test/java/com/example/extractor/MainTest.java]: Copied manually and Reverted changes to MainsTest.java. Test cases passing. 
```
Starting Gradle Daemon...
Gradle Daemon started in 3 s 490 ms

> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses
> Task :test
Generating all-in-one configuration...
Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example1.java
Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example2.java
Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example3.java
Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example4.java
Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example5.java
Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example6.java
Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example7.java
Loading configuration from example file...
Loading configuration from example file...
Loading configuration from example file...
Loading configuration from example file...
Loading configuration from example file...
Loading configuration from example file...
Loading configuration from example file...
MainTest > testGeneratedConfigMatchesExpectedAllInOne() PASSED
MainTest > testGeneratedConfigMatchesExpectedExample1() PASSED
MainTest > testGeneratedConfigMatchesExpectedExample2() PASSED
MainTest > testGeneratedConfigMatchesExpectedExample3() PASSED
MainTest > testGeneratedConfigMatchesExpectedExample4() PASSED
MainTest > testGeneratedConfigMatchesExpectedExample5() PASSED
MainTest > testGeneratedConfigMatchesExpectedExample6() PASSED
MainTest > testGeneratedConfigMatchesExpectedExample7() PASSED
MainTest > testParseConfigFromHeader() PASSED
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD SUCCESSFUL in 19s
5 actionable tasks: 3 executed, 2 up-to-date
12:35:57 am: Execution finished ':test --tests "com.example.extractor.MainTest"'.
```

File: extractor/src/test/resources/generated-config-all-in-one.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
```
Comment (romani) [inline extractor/src/test/resources/generated-config-all-in-one.xml:1]: Please in name of file "generated" to "expected"
And remove old expected.

**Review by romani — CHANGES_REQUESTED** [review]
Last

File: extractor/src/test/resources/generated-config-all-in-one.xml (hunk 2 above)
Comment (relentless-pursuit) [reply extractor/src/test/resources/generated-config-all-in-one.xml:1]: chane names manually right?

Comment (relentless-pursuit) [reply extractor/src/test/resources/generated-config-all-in-one.xml:1]: I reverted back the maintest.java where i was generating files. do you want me to revert maintest.java now?

Comment (relentless-pursuit) [reply extractor/src/test/resources/generated-config-all-in-one.xml]: Done.

File: extractor/src/test/resources/expected-config-all-in-one.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
```
Comment (romani) [inline extractor/src/test/resources/expected-config-all-in-one.xml:1]: Such files should substitute all files https://github.com/checkstyle/test-configs/tree/main/extractor/src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Configs

Please copy such files to this folder.

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: extractor/src/test/resources/expected-config-all-in-one.xml (hunk 3 above)
Comment (relentless-pursuit) [reply extractor/src/test/resources/expected-config-all-in-one.xml:1]: replaced all files in the above path

**Comment by romani** [comment]
Remove files at https://github.com/relentless-pursuit/test-configs/tree/PR5.2/extractor/src/test/resources
As you have them in subfolders.
There should be only one expected file for exampleX

**Comment by relentless-pursuit** [comment]
```
piyushkumarsadangi@piyushs-MacBook-Air test-configs % ls     
AbbreviationAsWordInName        SummaryJavadoc                  extractor
README.md                       checkstyle                      setup.sh
piyushkumarsadangi@piyushs-MacBook-Air test-configs % rm -rf AbbreviationAsWordInName 
piyushkumarsadangi@piyushs-MacBook-Air test-configs % cd extractor 
piyushkumarsadangi@piyushs-MacBook-Air extractor % ./gradlew test

Starting a Gradle Daemon, 1 busy and 1 incompatible Daemons could not be reused, use --status for details

> Task :test

MainTest > testGeneratedConfigMatchesExpectedAllInOne() STANDARD_OUT
    Generating all-in-one configuration...
    Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example1.java
    Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example2.java
    Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example3.java
    Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example4.java
    Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example5.java
    Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example6.java
    Loading configuration from example file: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInName/Examples/Example7.java

MainTest > testGeneratedConfigMatchesExpectedAllInOne() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample1() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample1() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample2() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample2() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample3() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample3() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample4() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample4() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample5() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample5() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample6() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample6() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample7() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample7() PASSED

MainTest > testParseConfigFromHeader() PASSED

MainsLauncherTest > testMain() STANDARD_OUT
    Current working directory: /Users/piyushkumarsadangi/Development/test-configs/extractor
    Walking through the input directory to collect Example#.java files...
    Found 7 Example#.java files.
    Output directory already exists: /Users/piyushkumarsadangi/Development/test-configs/extractor/AbbreviationAsWordInName
    Template file path: /Users/piyushkumarsadangi/Development/test-configs/extractor/build/resources/main/config-template-treewalker.xml
    Processing file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example4.java
    File content:
    /*xml
    <module name="Checker">
      <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">
          <property name="allowedAbbreviationLength" value="1"/>
          <property name="allowedAbbreviations" value="CSV"/>
          <property name="ignoreStatic" value="true"/>
          <property name="tokens" value="VARIABLE_DEF"/>
        </module>
      </module>
    </module>


    */

    package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;

    // xdoc section -- start
    class Example4 { // OK, ignore checking the class name
      int firstNum; // OK, abbreviation "N" is of allowed length 1
      int secondNUm;
      int secondMYNum; // violation 'no more than '2' consecutive capital letters'
      int thirdNUM; // violation 'no more than '2' consecutive capital letters'
      static int fourthNUM; // OK, variables with static modifier would be ignored
      String firstCSV; // OK, CSV abbreviation is allowed
      String firstXML; // violation 'no more than '2' consecutive capital letters'
      final int TOTAL = 5; // OK, final is ignored
      static final int LIMIT = 10; // OK, static final is ignored
    }
    // xdoc section -- end

    Generating configuration for file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example4.java
    Loading configuration from example file...
    Generated configuration:
    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
            "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
            "https://checkstyle.org/dtds/configuration_1_3.dtd">

    <module name = "Checker">
        <property name="charset" value="UTF-8"/>

        <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
        <property name="severity" value="warning"/>

        <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
        <property name="haltOnException" value="false"/>

        <!-- BeforeExecutionFileFilters is required for sources of java9 -->
        <module name="BeforeExecutionExclusionFileFilter">
            <property name="fileNamePattern" value="module\-info\.java$" />
        </module>

        <module name="TreeWalker">
            <module name="AbbreviationAsWordInName">
                <property name="allowedAbbreviationLength" value="1"/>
                <property name="allowedAbbreviations" value="CSV"/>
                <property name="ignoreStatic" value="true"/>
                <property name="tokens" value="VARIABLE_DEF"/>
            </module>

            <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
            <module name="SuppressionXpathSingleFilter">
                <property name="message" value="Javadoc comment at column \d+ has parse error"/>
            </module>
        </module>

        <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
        <module name="SuppressionSingleFilter">
            <property name="checks" value="Checker"/>
            <property name="message" value="Exception occurred while parsing"/>
        </module>
    </module>

    Writing generated configuration to: AbbreviationAsWordInName/Example4/Example4-config.xml
    Template file path: /Users/piyushkumarsadangi/Development/test-configs/extractor/build/resources/main/config-template-treewalker.xml
    Processing file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example5.java
    File content:
    /*xml
    <module name="Checker">
      <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">
          <property name="allowedAbbreviationLength" value="0"/>
          <property name="ignoreFinal" value="false"/>
          <property name="ignoreStatic" value="false"/>
          <property name="ignoreStaticFinal" value="true"/>
          <property name="tokens" value="VARIABLE_DEF"/>
        </module>
      </module>
    </module>


    */

    package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;

    // xdoc section -- start
    class Example5 {
      int counterXYZ = 1; // violation 'no more than '1' consecutive capital letters'
      // violation below 'no more than '1' consecutive capital letters'
      final int customerID = 2;
      static int nextID = 3; // violation 'no more than '1' consecutive capital letters'
      static final int MAX_ALLOWED = 4; // OK, ignored
    }
    // xdoc section -- end

    Generating configuration for file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example5.java
    Loading configuration from example file...
    Generated configuration:
    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
            "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
            "https://checkstyle.org/dtds/configuration_1_3.dtd">

    <module name = "Checker">
        <property name="charset" value="UTF-8"/>

        <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
        <property name="severity" value="warning"/>

        <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
        <property name="haltOnException" value="false"/>

        <!-- BeforeExecutionFileFilters is required for sources of java9 -->
        <module name="BeforeExecutionExclusionFileFilter">
            <property name="fileNamePattern" value="module\-info\.java$" />
        </module>

        <module name="TreeWalker">
            <module name="AbbreviationAsWordInName">
                <property name="allowedAbbreviationLength" value="0"/>
                <property name="ignoreFinal" value="false"/>
                <property name="ignoreStatic" value="false"/>
                <property name="ignoreStaticFinal" value="true"/>
                <property name="tokens" value="VARIABLE_DEF"/>
            </module>

            <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
            <module name="SuppressionXpathSingleFilter">
                <property name="message" value="Javadoc comment at column \d+ has parse error"/>
            </module>
        </module>

        <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
        <module name="SuppressionSingleFilter">
            <property name="checks" value="Checker"/>
            <property name="message" value="Exception occurred while parsing"/>
        </module>
    </module>

    Writing generated configuration to: AbbreviationAsWordInName/Example5/Example5-config.xml
    Template file path: /Users/piyushkumarsadangi/Development/test-configs/extractor/build/resources/main/config-template-treewalker.xml
    Processing file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example2.java
    File content:
    /*xml
    <module name="Checker">
      <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">
          <property name="ignoreStatic" value="false"/>
          <property name="ignoreOverriddenMethods" value="false"/>
        </module>
      </module>
    </module>


    */

    package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;

    import java.util.HashSet;
    import java.util.Set;

    // xdoc section -- start
    class Example2 extends SuperClass { // OK, camel case
      int CURRENT_COUNTER; // violation 'no more than '4' consecutive capital letters'
      // violation below 'no more than '4' consecutive capital letters'
      static int GLOBAL_COUNTER;
      final Set<String> stringsFOUND = new HashSet<>(); // OK, final is ignored

      @Override // violation below 'no more than '4' consecutive capital letters'
      public void printCOUNTER() {
        System.out.println(CURRENT_COUNTER); // OK, only definitions are checked
      }
      // violation below 'no more than '4' consecutive capital letters'
      void incrementCOUNTER() {
        CURRENT_COUNTER++; // OK, only definitions are checked
      }
      // violation below 'no more than '4' consecutive capital letters'
      static void incrementGLOBAL() {
        GLOBAL_COUNTER++; // OK, only definitions are checked
      }
    }
    // xdoc section -- end

    Generating configuration for file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example2.java
    Loading configuration from example file...
    Generated configuration:
    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
            "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
            "https://checkstyle.org/dtds/configuration_1_3.dtd">

    <module name = "Checker">
        <property name="charset" value="UTF-8"/>

        <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
        <property name="severity" value="warning"/>

        <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
        <property name="haltOnException" value="false"/>

        <!-- BeforeExecutionFileFilters is required for sources of java9 -->
        <module name="BeforeExecutionExclusionFileFilter">
            <property name="fileNamePattern" value="module\-info\.java$" />
        </module>

        <module name="TreeWalker">
            <module name="AbbreviationAsWordInName">
                <property name="ignoreOverriddenMethods" value="false"/>
                <property name="ignoreStatic" value="false"/>
            </module>

            <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
            <module name="SuppressionXpathSingleFilter">
                <property name="message" value="Javadoc comment at column \d+ has parse error"/>
            </module>
        </module>

        <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
        <module name="SuppressionSingleFilter">
            <property name="checks" value="Checker"/>
            <property name="message" value="Exception occurred while parsing"/>
        </module>
    </module>

    Writing generated configuration to: AbbreviationAsWordInName/Example2/Example2-config.xml
    Template file path: /Users/piyushkumarsadangi/Development/test-configs/extractor/build/resources/main/config-template-treewalker.xml
    Processing file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example3.java
    File content:
    /*xml
    <module name="Checker">
      <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">
          <property name="allowedAbbreviationLength" value="0"/>
          <property name="allowedAbbreviations" value="XML,URL,O"/>
          <property name="ignoreStatic" value="false"/>
          <property name="tokens" value="VARIABLE_DEF,CLASS_DEF"/>
        </module>
      </module>
    </module>


    */

    package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;

    // xdoc section -- start
    class Example3 {
      int firstNum;
      int secondNUM; // violation 'no more than '1' consecutive capital letters'
      static int thirdNum; // OK, the static modifier would be checked
      static int fourthNUm; // violation 'no more than '1' consecutive capital letters'
      String firstXML; // OK, XML abbreviation is allowed
      String firstURL; // OK, URL abbreviation is allowed
      final int TOTAL = 5; // OK, final is ignored
      static final int LIMIT = 10; // OK, static final is ignored
      void newOAuth2Client() {} // OK, O abbreviation is allowed
      void OAuth2() {} // OK, O abbreviation is allowed
      void OAUth2() {}
    }
    // xdoc section -- end

    Generating configuration for file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example3.java
    Loading configuration from example file...
    Generated configuration:
    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
            "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
            "https://checkstyle.org/dtds/configuration_1_3.dtd">

    <module name = "Checker">
        <property name="charset" value="UTF-8"/>

        <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
        <property name="severity" value="warning"/>

        <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
        <property name="haltOnException" value="false"/>

        <!-- BeforeExecutionFileFilters is required for sources of java9 -->
        <module name="BeforeExecutionExclusionFileFilter">
            <property name="fileNamePattern" value="module\-info\.java$" />
        </module>

        <module name="TreeWalker">
            <module name="AbbreviationAsWordInName">
                <property name="allowedAbbreviationLength" value="0"/>
                <property name="allowedAbbreviations" value="XML,URL,O"/>
                <property name="ignoreStatic" value="false"/>
                <property name="tokens" value="VARIABLE_DEF,CLASS_DEF"/>
            </module>

            <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
            <module name="SuppressionXpathSingleFilter">
                <property name="message" value="Javadoc comment at column \d+ has parse error"/>
            </module>
        </module>

        <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
        <module name="SuppressionSingleFilter">
            <property name="checks" value="Checker"/>
            <property name="message" value="Exception occurred while parsing"/>
        </module>
    </module>

    Writing generated configuration to: AbbreviationAsWordInName/Example3/Example3-config.xml
    Template file path: /Users/piyushkumarsadangi/Development/test-configs/extractor/build/resources/main/config-template-treewalker.xml
    Processing file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example1.java
    File content:
    /*xml
    <module name="Checker">
      <module name="TreeWalker">
        <module name="AbbreviationAsWordInName"/>
      </module>
    </module>


    */

    package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;

    import java.util.HashSet;
    import java.util.Set;

    // xdoc section -- start
    class Example1 extends SuperClass { // OK, camel case
      int CURRENT_COUNTER; // violation 'no more than '4' consecutive capital letters'
      static int GLOBAL_COUNTER; // OK, static is ignored
      final Set<String> stringsFOUND = new HashSet<>(); // OK, final is ignored

      @Override
      public void printCOUNTER() { // OK, overridden method is ignored
        System.out.println(CURRENT_COUNTER);
      }
      // violation below 'no more than '4' consecutive capital letters'
      void incrementCOUNTER() {
        CURRENT_COUNTER++; // OK, only definitions are checked
      }
      // violation below 'no more than '4' consecutive capital letters'
      static void incrementGLOBAL() {
        GLOBAL_COUNTER++; // OK, only definitions are checked
      }
    }
    // xdoc section -- end

    Generating configuration for file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example1.java
    Loading configuration from example file...
    Generated configuration:
    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
            "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
            "https://checkstyle.org/dtds/configuration_1_3.dtd">

    <module name = "Checker">
        <property name="charset" value="UTF-8"/>

        <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
        <property name="severity" value="warning"/>

        <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
        <property name="haltOnException" value="false"/>

        <!-- BeforeExecutionFileFilters is required for sources of java9 -->
        <module name="BeforeExecutionExclusionFileFilter">
            <property name="fileNamePattern" value="module\-info\.java$" />
        </module>

        <module name="TreeWalker">
            <module name="AbbreviationAsWordInName"/>

            <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
            <module name="SuppressionXpathSingleFilter">
                <property name="message" value="Javadoc comment at column \d+ has parse error"/>
            </module>
        </module>

        <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
        <module name="SuppressionSingleFilter">
            <property name="checks" value="Checker"/>
            <property name="message" value="Exception occurred while parsing"/>
        </module>
    </module>

    Writing generated configuration to: AbbreviationAsWordInName/Example1/Example1-config.xml
    Template file path: /Users/piyushkumarsadangi/Development/test-configs/extractor/build/resources/main/config-template-treewalker.xml
    Processing file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example6.java
    File content:
    /*xml
    <module name="Checker">
      <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">
          <property name="allowedAbbreviationLength" value="0"/>
          <property name="ignoreFinal" value="false"/>
          <property name="ignoreStatic" value="true"/>
          <property name="ignoreStaticFinal" value="false"/>
          <property name="tokens" value="VARIABLE_DEF"/>
        </module>
      </module>
    </module>


    */

    package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;

    // xdoc section -- start
    class Example6 {
      int counterXYZ = 1; // violation 'no more than '1' consecutive capital letters'
      // violation below 'no more than '1' consecutive capital letters'
      final int customerID = 2;
      static int nextID = 3; // OK, ignored
      // violation below 'no more than '1' consecutive capital letters'
      static final int MAX_ALLOWED = 4;
    }
    // xdoc section -- end

    Generating configuration for file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example6.java
    Loading configuration from example file...
    Generated configuration:
    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
            "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
            "https://checkstyle.org/dtds/configuration_1_3.dtd">

    <module name = "Checker">
        <property name="charset" value="UTF-8"/>

        <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
        <property name="severity" value="warning"/>

        <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
        <property name="haltOnException" value="false"/>

        <!-- BeforeExecutionFileFilters is required for sources of java9 -->
        <module name="BeforeExecutionExclusionFileFilter">
            <property name="fileNamePattern" value="module\-info\.java$" />
        </module>

        <module name="TreeWalker">
            <module name="AbbreviationAsWordInName">
                <property name="allowedAbbreviationLength" value="0"/>
                <property name="ignoreFinal" value="false"/>
                <property name="ignoreStatic" value="true"/>
                <property name="ignoreStaticFinal" value="false"/>
                <property name="tokens" value="VARIABLE_DEF"/>
            </module>

            <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
            <module name="SuppressionXpathSingleFilter">
                <property name="message" value="Javadoc comment at column \d+ has parse error"/>
            </module>
        </module>

        <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
        <module name="SuppressionSingleFilter">
            <property name="checks" value="Checker"/>
            <property name="message" value="Exception occurred while parsing"/>
        </module>
    </module>

    Writing generated configuration to: AbbreviationAsWordInName/Example6/Example6-config.xml
    Template file path: /Users/piyushkumarsadangi/Development/test-configs/extractor/build/resources/main/config-template-treewalker.xml
    Processing file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java
    File content:
    /*xml
    <module name="Checker">
      <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">
          <property name="allowedAbbreviations" value="ALLOWED"/>
          <property name="ignoreStaticFinal" value="false"/>
        </module>
      </module>
    </module>


    */

    package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;

    // xdoc section -- start
    class Example7 {
      int counterXYZ = 1;
      final int customerID = 2;
      static int nextID = 3;
      static final int MAX_ALLOWED = 4; // OK, abbreviation is allowed
    }
    // xdoc section -- end

    Generating configuration for file: .ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java
    Loading configuration from example file...
    Generated configuration:
    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
            "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
            "https://checkstyle.org/dtds/configuration_1_3.dtd">

    <module name = "Checker">
        <property name="charset" value="UTF-8"/>

        <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
        <property name="severity" value="warning"/>

        <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
        <property name="haltOnException" value="false"/>

        <!-- BeforeExecutionFileFilters is required for sources of java9 -->
        <module name="BeforeExecutionExclusionFileFilter">
            <property name="fileNamePattern" value="module\-info\.java$" />
        </module>

        <module name="TreeWalker">
            <module name="AbbreviationAsWordInName">
                <property name="allowedAbbreviations" value="ALLOWED"/>
                <property name="ignoreStaticFinal" value="false"/>
            </module>

            <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
            <module name="SuppressionXpathSingleFilter">
                <property name="message" value="Javadoc comment at column \d+ has parse error"/>
            </module>
        </module>

        <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
        <module name="SuppressionSingleFilter">
            <property name="checks" value="Checker"/>
            <property name="message" value="Exception occurred while parsing"/>
        </module>
    </module>

    Writing generated configuration to: AbbreviationAsWordInName/Example7/Example7-config.xml

MainsLauncherTest > testMain() PASSED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 4s
5 actionable tasks: 1 executed, 4 up-to-date
piyushkumarsadangi@piyushs-MacBook-Air extractor % ls
AbbreviationAsWordInName        build.gradle                    gradlew                         settings.gradle
build                           gradle                          gradlew.bat                     src
piyushkumarsadangi@piyushs-MacBook-Air extractor % cd AbbreviationAsWordInName 
piyushkumarsadangi@piyushs-MacBook-Air AbbreviationAsWordInName % ls
Example1        Example2        Example3        Example4        Example5        Example6        Example7
piyushkumarsadangi@piyushs-MacBook-Air AbbreviationAsWordInName % cd Example
cd: no such file or directory: Example
piyushkumarsadangi@piyushs-MacBook-Air AbbreviationAsWordInName % cd Example1
piyushkumarsadangi@piyushs-MacBook-Air Example1 % ls
Example1-config.xml
piyushkumarsadangi@piyushs-MacBook-Air Example1 % 

```

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
@relentless-pursuit , please fix CI https://github.com/checkstyle/test-configs/actions/runs/9564822716/job/26366421850?pr=38#step:5:52
it might be again some problem with path, keep in might that Linux is case sensitive, path must be exact.

**Comment by relentless-pursuit** [comment]
> @relentless-pursuit , please fix CI https://github.com/checkstyle/test-configs/actions/runs/9564822716/job/26366421850?pr=38#step:5:52 it might be again some problem with path, keep in might that Linux is case sensitive, path must be exact.

Done. CI passed.

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
We do progress, nice!

**Pull request merged**